stdlib/2/os/path.pyi

```diff
@@ -22,25 +28,32 @@ devnull = ...  # type: str
 def abspath(path: AnyStr) -> AnyStr: ...
 def basename(path: AnyStr) -> AnyStr: ...
 
-def commonprefix(list: List[AnyStr]) -> AnyStr: ...
+if sys.version_info >= (3, 5):
+    def commonpath(paths: Sequence[AnyStr]) -> AnyStr: ...
+
+# NOTE: Empty lists results in '' (str) regardless of contained type.
+# Also, in Python 2 mixed sequences of Text and bytes results in either Text or bytes
```

Comment:
`Sequence[AnyStr]` works in python 2 because `str` gets promoted to `Text`


And sadly I can't overload where the argument varies by what's inside the Sequence.
Basically if I do in Python 2:
```
@overload
def commonprefix(list: Sequence[str]) -> str: ...
@overload
def commonprefix(list: Sequence[Text]) -> Any: ...
```

I get this error:
`error: Overloaded function signatures 1 and 2 overlap with incompatible return types`

It doesn't work with List[X] or with Sequence[bytes] in python 3.  I even tried introducing a `_Str = TypeVar("_Str", str)` and separating using that, but that doesn't work either.

So I opted for the most generic thing.